mountpoint-s3/src/fs.rs

```diff
@@ -288,8 +289,15 @@ where
             }
         } else {
             lookup.inode.start_reading()?;
+            let file_etag = ETag {
+                etag: match lookup.stat.etag {
+                    None => return Err(libc::EBADF),
+                    Some(etag) => etag,
+                },
+            };
```

Comment:
I need to think about this a bit more. In what cases do we expect a lookup not to have an etag?